jerry-core/vm/vm.c

```diff
@@ -1651,6 +1657,36 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
         }
         case VM_OC_ADD:
         {
+          if (ecma_are_values_integer_numbers (left_value, right_value))
```

Comment:
For the newly added code to the VM_OC_ADD and VM_OC_SUB cases there seems to be a lot of code duplication (minus the actual add/sub operations and invocations of the respective helper functions). This probably could be factored out into a static inline helper function without a negative performance impact. Not sure it's worth the hassle though, I guess it's up to you whether you want to go that route :)